lib/facility.rb

```diff
@@ -1,14 +1,87 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info, collected_fees = 0)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = collected_fees
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
+    
   end
+
+  def register_vehicle(vehicle)
+    if available_service?("Vehicle Registration")
```

Comment:
You could flatten the conditional below by making this line into a guard clause, which will exit the method early if a certain condition isn't met.
```
unavailable_msg = "This facility does not offer vehicle registration services."
return unavailable_msg if available_service?("Vehicle Registration")
```